src/main/java/Duke.java

```diff
@@ -6,5 +14,167 @@ public static void main(String[] args) {
+
+        while (runLoop) {
+            String userInput = sc.nextLine().trim();
+
+            if (userInput.equalsIgnoreCase("bye")) {
+                exitProgram();
+                runLoop = false;
+            } else if (userInput.equalsIgnoreCase("list")) {
+                listTasks(tasks);
+            } else if (userInput.split(" ")[0].equalsIgnoreCase("done")) {
+                markTask(userInput, tasks);
+            }
+            else if (userInput.split(" ")[0].equalsIgnoreCase("delete")) {
+                deleteTask(userInput, tasks);
+            }
+            else {
+                addTask(userInput, tasks);
+            }
+        }
+    }
+
+    public static void exitProgram() {
+        // exits the program
+        System.out.println("\t------------------------------------------");
+        System.out.println("\tSee you soon! Goodbye! ^.^");
+        System.out.println("\t__________________________________________\n");
+    }
+
+    public static void listTasks(ArrayList<Task> tasks) {
+        System.out.println("\t------------------------------------------");
+        if (tasks.size() == 0) {
+            // if list is empty
+            System.out.println("\tYour list is empty!");
+        }
+        else {
+            // displays the list of tasks
+            System.out.println("\tHere are the tasks in your list: ");
+            int i = 1;
+            for (Task task : tasks) {
+                System.out.println("\t" + i++ + "." + task.toString());
+            }
+        }
+        System.out.println("\t__________________________________________\n");
+    }
+
+    public static void markTask(String userInput, ArrayList<Task> tasks) throws NumberFormatException,
+            IndexOutOfBoundsException {
+        // check if task exists
+        System.out.println("\t------------------------------------------");
+        try {
+            int taskIndex = Integer.parseInt(userInput.split(" ")[1]) - 1;
+            if (taskIndex > tasks.size()-1) {
+                System.out.println("\tTask " + ++taskIndex + " does not exist! Please try again.");
+            }
+            else {
+                // sets a task as done
+                Task task = tasks.get(taskIndex);
+                task.markAsDone();
+                System.out.println("\tGreat job! I've marked this task as done: ");
+                System.out.println("\t" + task.toString());
+            }
+        }
+        catch (NumberFormatException e) {
+            System.out.println("\tOOPS!! To mark task, you have to enter an integer following the work done " +
+                    "in this format e.g. 'done 3'.");
+        }
+        catch (IndexOutOfBoundsException e) {
+            System.out.println("\tOOPS!! Invalid task input!");
+        }
+        System.out.println("\t__________________________________________\n");
+    }
+
+    public static void addTask(String userInput, ArrayList<Task> tasks) throws StringIndexOutOfBoundsException {
```

Comment:
This method is really long; consider breaking it down into smaller methods that would also make it easier for you to debug your own work. Maybe, upon doing so, you could move some of those methods to their relevant class (i.e. Deadline class, Event class, etc) to enhance code organisation